src/CommunityToolkit.Maui/PopupService.cs

```diff
@@ -63,7 +63,7 @@ public void ShowPopup<TViewModel>(TViewModel viewModel) where TViewModel : INoti
 
 		var popup = GetPopup(typeof(TViewModel));
 
-		ValidateBindingContext<TViewModel>(popup, out _);
```

Comment:
I think we should make sure that the BindingContext is not assigned before assigning it here to make sure we aren't replacing something already created. Mostly to help the developer know if they are making a mistake. What do you think?

Reply:
I agree with you and have modified the code

Reply:
But isn't setting the `BindingContext` of the page in any case exactly the purpose of this method?
Imagine having a popup where the ViewModel is resolved using DI in most cases.
But for one specific case, we are manually resolving or creating the instance and want to pass it to the popup service.
The instance would never be set as `BindingContext` with this implementation.

Reply:
That is correct but I think it best to bring this to the developers attention (potentially with an exception). If they wanted to use your scenario they could potentially register their view models as `AddScoped` and share the instance...